src/open_prime_rando/dol_patching/echoes/dol_patches.py

```diff
@@ -500,3 +509,40 @@ def apply_map_door_changes(door_symbols: MapDoorTypeAddresses, dol_file: DolFile
 
     dol_file.symbols["CTweakAutoMapper::GetDoorColor::DoorColorArray"] = door_color_array
     dol_file.write("CTweakAutoMapper::GetDoorColor::DoorColorArray", DoorMapIcon.get_surface_colors_as_bytes())
+
+
+def apply_widescreen_hack(widescreen_render_symbols: WidescreenRenderAddresses, dol_file: DolFile, enabled: bool):
+    """
+    Apply widescreen render hack to render the game in 16:9
+    """
+    culling_replacement = widescreen_render_symbols.culling_replacement
+    culling_insertion = widescreen_render_symbols.culling_insertion
+    viewport_replacement = widescreen_render_symbols.viewport_replacement
+    viewport_insertion = widescreen_render_symbols.viewport_insertion
+
+    if enabled:
+        dol_file.write_instructions(culling_replacement, [bl(culling_insertion, relative=False)])
+        dol_file.write_instructions(culling_insertion, [lis(r14, 0x4000), stw(r14, 0, r2), lfs(f26, 0, r2), blr()])
+
+        dol_file.write_instructions(viewport_replacement, [b(viewport_insertion, relative=False)])
+        dol_file.write_instructions(
+            viewport_insertion,
+            [
+                lis(r14, 0x3FAA),
```

Comment:
Can you document a bit excessively what these instructions are supposed to do, and what you are patching?

Think of having to come back to this code in 3 years.